cfme/tests/intelligence/test_chargeback.py

```diff
@@ -68,42 +67,49 @@ def chargeback_storage_rate(rate_type='fixed'):
         })
 
 
-def test_add_new_compute_chargeback(chargeback_compute_rate):
-    chargeback_compute_rate.create()
-    view = chargeback_compute_rate.create_view(
-        navigator.get_class(chargeback_compute_rate, 'All').VIEW)
-    view.flash.assert_success_message('Chargeback Rate "{}" was added'
-                                      .format(chargeback_compute_rate.description))
-    assert chargeback_compute_rate.exists
+@pytest.mark.tier(3)
+def test_compute_chargeback_duplicate_disallowed(request):
+    cb_rate = chargeback_compute_rate('fixed')
 
+    @request.addfinalizer
+    def _cleanup():
+        if cb_rate.exists:
+            cb_rate.delete()
 
-@pytest.mark.tier(3)
-def test_compute_chargeback_duplicate_disallowed(chargeback_compute_rate):
-    chargeback_compute_rate.create()
-    assert chargeback_compute_rate.exists
+    cb_rate.create()
+    assert cb_rate.exists
     with pytest.raises(AssertionError):
-        chargeback_compute_rate.create()
+        cb_rate.create()
     # view should still be on the add form
-    view = chargeback_compute_rate.create_view(
-        navigator.get_class(chargeback_compute_rate, 'Add').VIEW)
+    view = cb_rate.create_view(
+        navigator.get_class(cb_rate, 'Add').VIEW)
     view.flash.assert_message('Description has already been taken', t='error')
     # cancel form, check all redirect
     view.cancel_button.click()
-    view = chargeback_compute_rate.create_view(
-        navigator.get_class(chargeback_compute_rate, 'All').VIEW)
+    view = cb_rate.create_view(
+        navigator.get_class(cb_rate, 'All').VIEW)
     assert view.is_displayed
     view.flash.assert_success_message('Add of new Chargeback Rate was cancelled by the user')
 
 
 @pytest.mark.tier(3)
-@pytest.mark.parametrize('storage_chargeback',
-                         ['fixed', 'variable'],
-                         ids=['fixed_storage_rate', 'variable_storage_rate'])
-@pytest.mark.meta(blockers=[
-    BZ(1532368, forced_streams='5.9',
-       unblock=lambda storage_chargeback: 'variable' not in storage_chargeback)])
-def test_add_new_storage_chargeback(storage_chargeback, request):
-    cb_rate = chargeback_storage_rate(storage_chargeback)
+@pytest.mark.parametrize('chargeback_rate',
+                         ['compute-fixed-add', 'compute-variable-add', 'storage-fixed-add',
```

Comment:
Did you consider parametrizing these components separately instead of as one long list?  The ids are so close to the actual parameters, it seems they could be generated in a list comprehension with a `format()` call.

This is just a bit difficult to read and parse out, since compute/storage and fixed/variable are mixed up, and they're organized by the action, which is the last field.

something like
```
@pytest.mark.parametrize('rate_resource', ['compute', 'storage'])
@pytest.mark.parametrize('rate_type', ['fixed', 'variable'])
@pytest.mark.parametrize('rate_action', ['add', 'delete', 'edit'])
```

Then call the test `test_chargeback_rates` instead of trying to put 'rate' in the parameter ID.

Reply:
If you split parametrization like this, you can use the `chargeback_[storage|compute]_rate` method as a fixture that references rate_resource, and then do yield/cleanup there instead of within the test.  Just something to consider.

Reply:
Why split add/edit/delete into separate parametrization, instead of the test method doing create/edit/delete on both types  (and both resources) as a single CRUD test? This would remove one layer of parametrization - effectively this still makes for as many tests as there were before, just combined into a single method with if blocks.

Reply:
@mshriver , I've turned chargeback_rate into a fixture now and the cleanup is happening in the fixture.

I have add/edit/delete as separate tests since those operations are available as separate options on the UI.